contracts/TokenHolder.sol

```diff
@@ -90,6 +99,38 @@ contract TokenHolder is MultiSigWallet {
         )
     );
 
+    /**
+     *  TH redeem function call prefix needed to verify signed data as per
+     *  EIP1077 proposal.
+     */
+    bytes4 public constant REDEEM_RULE_CALLPREFIX = bytes4(
+        keccak256(
+            "redeem(uint256,address,uint256,uint256,uint256,bytes32,uint256,uint8,bytes32,bytes32)"
+        )
+    );
+
+    /**
+     *  Not all coGateway.redeem parameters are signed. HashLock and
+     *  facilitator are not part of data which ephemeral key is signing as this
+     *  information is not known by the user at the time of signing.
+     *  This callPrefix is needed to recover the ephemeral key.
+     */
+    bytes4 public constant REDEEM_SIGNED_DATA_CALLPREFIX = bytes4(
```

Comment:
We discussed about this in wiki. Row No. 10 in the table
https://ostdotcom.atlassian.net/wiki/spaces/OP/pages/506626049/Redeem+RevertRedemption

data to be signed is inspired from CoGateway.redeem function signature: CoGateway.redeem(
uint256 _amount,
address _beneficiary,
address _facilitator,
uint256 _gasPrice,
uint256 _gasLimit,
uint256 _nonce,
bytes32 _hashLock
)

Except _hashLock and _facilitator every other data point will be available at the time of signing. HashLock is not part of signed data. Facilitator argument will be removed altogether from CoGateway.redeem. There is already a ticket for it. 

So we are left with _amount, _beneficiary, _gasPrice, _gasLimit, _nonce.

All above data can be signed by user Ekey as user has information about it. Having multiple data points for signing makes the signed data robust. 

I have added below documentation:
"/**
     *  It's needed because not all coGateway.redeem parameters are signed.
     *  HashLock and facilitator are not part of data which ephemeral key is
     *  signing as this information is not known by the user at the time of
     *  signing. This callPrefix is needed to recover the ephemeral key.
     */"
 
Let me know if it's still not clear.